src/logic/safe/store/reducer/pendingTransactions.ts

```diff
@@ -0,0 +1,33 @@
+import { Action, handleActions } from 'redux-actions'
+
+import session from 'src/utils/storage/session'
+import { PENDING_TRANSACTIONS_ACTIONS } from 'src/logic/safe/store/actions/pendingTransactions'
+
+type SafeTxHash = string
+export type PendingTransactionsState = Set<SafeTxHash>
+
+export type PendingTransactionPayload = {
+  safeTxHash: string
+  isBroadcast?: boolean
+}
+
+export const PENDING_TRANSACTIONS_ID = 'pendingTransactions'
+
+export const pendingTransactionsReducer = handleActions<PendingTransactionsState, PendingTransactionPayload>(
+  {
+    [PENDING_TRANSACTIONS_ACTIONS.ADD]: (
+      state: PendingTransactionsState,
+      action: Action<PendingTransactionPayload>,
+    ) => {
+      return state.add(action.payload.safeTxHash)
+    },
+    [PENDING_TRANSACTIONS_ACTIONS.REMOVE]: (
+      state: PendingTransactionsState,
+      action: Action<PendingTransactionPayload>,
+    ) => {
+      state.delete(action.payload.safeTxHash)
```

Comment:
Reducers are supposed to be pure functions. You're mutating the set here.
You can either use the `Set` from immutable.js, or make a shallow copy like this:
`new Set(state)`